semgrep-core/tests/tainting_rules/ts/2787-fun.ts

```diff
@@ -0,0 +1,16 @@
+// https://github.com/returntocorp/semgrep/issues/2787
+import React, { Component } from "react";
+class BadPractice extends Component {
+  dangerous() {
```

Comment:
So if this method is defined after render, it will not detect the tainting?
The order matters right?

Reply:
Yes, as of now, that is correct. We could iterate the methods in topological order wrt their dependencies.